frontend/docsettings.lua

```diff
@@ -75,6 +91,14 @@ function DocSettings:getSidecarDir(doc_path, force_location)
     local location = force_location or G_reader_settings:readSetting("document_metadata_folder", "doc")
     if location == "dir" then
         path = DOCSETTINGS_DIR..path
+    elseif location == "hash" then
+        local file = io.open(doc_path, 'rb')
+        if not file then return "" end
```

Comment:
If a hash cannot be calculated we can take some substitution for a folder name, maybe just `path`, same as for "dir" option.

Reply:
Maybe I'll comment `getSidecarDir` to explain that `force_location` prefers that location _if available_ and otherwise defaults to what it can find. I don't think this has implications on any of the code I added though it does make some calls like https://github.com/ryanwwest/koreader/blob/4c858113445eb008755947c53bd184dd5c4bd76a/frontend/docsettings.lua#L147 a bit confusing. Maybe not a problem though.